app/frontend/src/components/ConfirmationDialogWrapper.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { cloneElement, ReactElement, useState } from "react";
+
+import Button from "./Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "./Dialog";
+
+interface ConfirmationDialogWrapperProps<P extends { onClick: () => void }> {
```

Comment:
I was hoping being clever like this would mean TS would check the child component has "onClick", but it's not that smart (even if it's in a prop and not children), so I did the assert below

Reply:
In case it works in a future TS version? I think it will work when I change to render prop anyway